packages/anoncreds/src/utils/proofRequest.ts

```diff
@@ -9,16 +9,30 @@ import {
 
 export function proofRequestUsesUnqualifiedIdentifiers(proofRequest: AnonCredsProofRequest) {
   // We assume that if any identifier is unqualified, all of them are unqualified as well
-  return Object.values(proofRequest.requested_attributes).some((attribute) =>
-    attribute.restrictions?.some(
-      (restriction) =>
-        (restriction.cred_def_id && isUnqualifiedCredentialDefinitionId(restriction.cred_def_id)) ||
-        (restriction.schema_id && isUnqualifiedSchemaId(restriction.schema_id)) ||
-        (restriction.issuer_did && isUnqualifiedIndyDid(restriction.issuer_did)) ||
-        (restriction.issuer_id && isUnqualifiedIndyDid(restriction.issuer_id)) ||
-        (restriction.schema_issuer_did && isUnqualifiedIndyDid(restriction.schema_issuer_did)) ||
-        (restriction.schema_issuer_id && isUnqualifiedIndyDid(restriction.schema_issuer_id)) ||
-        (restriction.rev_reg_id && isUnqualifiedRevocationRegistryId(restriction.rev_reg_id))
+  return (
+    Object.values(proofRequest.requested_attributes).some((attribute) =>
+      attribute.restrictions?.some(
+        (restriction) =>
+          (restriction.cred_def_id && isUnqualifiedCredentialDefinitionId(restriction.cred_def_id)) ||
+          (restriction.schema_id && isUnqualifiedSchemaId(restriction.schema_id)) ||
+          (restriction.issuer_did && isUnqualifiedIndyDid(restriction.issuer_did)) ||
+          (restriction.issuer_id && isUnqualifiedIndyDid(restriction.issuer_id)) ||
+          (restriction.schema_issuer_did && isUnqualifiedIndyDid(restriction.schema_issuer_did)) ||
+          (restriction.schema_issuer_id && isUnqualifiedIndyDid(restriction.schema_issuer_id)) ||
+          (restriction.rev_reg_id && isUnqualifiedRevocationRegistryId(restriction.rev_reg_id))
+      )
+    ) ||
+    Object.values(proofRequest.requested_predicates).some((predicate) =>
+      predicate.restrictions?.some(
+        (restriction) =>
+          (restriction.cred_def_id && isUnqualifiedCredentialDefinitionId(restriction.cred_def_id)) ||
+          (restriction.schema_id && isUnqualifiedSchemaId(restriction.schema_id)) ||
+          (restriction.issuer_did && isUnqualifiedIndyDid(restriction.issuer_did)) ||
+          (restriction.issuer_id && isUnqualifiedIndyDid(restriction.issuer_id)) ||
+          (restriction.schema_issuer_did && isUnqualifiedIndyDid(restriction.schema_issuer_did)) ||
+          (restriction.schema_issuer_id && isUnqualifiedIndyDid(restriction.schema_issuer_id)) ||
+          (restriction.rev_reg_id && isUnqualifiedRevocationRegistryId(restriction.rev_reg_id))
+      )
```

Comment:
How about:


```suggestion
   [...Object.values(proofRequest.requested_attributes), ...Object.values(proofRequest.requested_predicates)].some((attribute) =>
      attribute.restrictions?.some(
        (restriction) =>
          (restriction.cred_def_id && isUnqualifiedCredentialDefinitionId(restriction.cred_def_id)) ||
          (restriction.schema_id && isUnqualifiedSchemaId(restriction.schema_id)) ||
          (restriction.issuer_did && isUnqualifiedIndyDid(restriction.issuer_did)) ||
          (restriction.issuer_id && isUnqualifiedIndyDid(restriction.issuer_id)) ||
          (restriction.schema_issuer_did && isUnqualifiedIndyDid(restriction.schema_issuer_did)) ||
          (restriction.schema_issuer_id && isUnqualifiedIndyDid(restriction.schema_issuer_id)) ||
          (restriction.rev_reg_id && isUnqualifiedRevocationRegistryId(restriction.rev_reg_id))
      )
```

Reply:
Love it, thanks for the rec. Made the change and also added to and updated the relevant e2e test suite.